File: leetcode/3-LongestSubstringWithoutRepeatingCharacters/python/main.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        temp = ''
        l = 0
        for c in s:
            if c not in temp:
                temp += c
                l = len(temp) if len(temp) > length else length
            else:
                temp = temp.split(c)[-1]
                temp += c
        return l


import unittest


class TestSolution(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def test_case_1(self):
        result = self.solution.lengthOfLongestSubstring('abcabcbb')
        self.assertEqual(result, 3)

    def test_case_2(self):
        result = self.solution.lengthOfLongestSubstring('bbbbb')
        self.assertEqual(result, 1)

    def test_case_3(self):
        result = self.solution.lengthOfLongestSubstring('pwwkew')
        self.assertEqual(result, 3)


if __name__ == '__main__':
    unittest.main()

File: leetcode/3-LongestSubstringWithoutRepeatingCharacters/go/main.go
package longestsubstring

import "strings"

func lengthOfLongestSubstring(s string) int {
	var (
		temp string
		l    int
	)

	for _, c := range strings.Split(s, "") {
		if strings.Contains(temp, c) {
			temp = strings.SplitN(temp, c, 2)[1]
			temp = temp + c
		} else {
			temp = temp + c
			if len(temp) > l {
				l = len(temp)
			}
		}
	}

	return l
}
